ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/netty/NettyP2PNetwork.java

```diff
@@ -268,8 +266,15 @@ protected void initChannel(final SocketChannel ch) {
                 connection.disconnect(DisconnectReason.TOO_MANY_PEERS);
                 return;
               }
+
               // Reject incoming connections that are blacklisted
-              if (peerBlacklist.contains(connection)) {
+              //   OR not whitelisted (if --node-whitelist)
+              if (peerBlacklist.contains(connection)
+                  || !nodeWhitelist.contains(
```

Comment:
I think we can make this statement easier to read (and remov the comments). Something like this:
```
boolean isPeerBlacklisted = peerBlacklist.contains(connection);
boolean isPeerNotWhitelisted = !isPeerWhitelisted(connection, ch);

if (isPeerBlacklisted || isPeerNotWhitelisted) {
    ....
}

...

private isPeerWhitelisted(...) {
return nodeWhitelist.contains(
                        connection.getPeer().getNodeId(),
                        ch.remoteAddress().getAddress().getHostAddress(),
                        connection.getPeer().getPort(),
                        connection.getPeer().getPort())
);
```

Reply:
One more thing... Do we need to check if the whitelist has been set before checking if the node is in the whitelist?

Reply:
the contains() method checks that. In PR #476 @mark-terry has refactored it to be called isPermitted() which I think makes more sense.